src/components/calcite-shell-panel/calcite-shell-panel.tsx

```diff
@@ -86,25 +142,222 @@ export class CalciteShellPanel {
+      >
         {this.renderHeader()}
         <div class={CSS.contentBody}>
           <slot />
         </div>
       </div>
     );
 
+    const separatorNode = resizable ? (
+      <div
+        aria-label={intlResize}
+        aria-orientation="horizontal"
+        aria-valuemax={contentWidthMax}
+        aria-valuemin={contentWidthMin}
+        aria-valuenow={contentWidth ?? initialContentWidth}
+        class={CSS.separator}
+        onKeyDown={this.separatorKeyDown}
+        ref={this.connectSeparator}
+        role="separator"
+        tabIndex={0}
+        touch-action="none"
+      />
+    ) : null;
+
     const actionBarNode = <slot name={SLOTS.actionBar} />;
 
-    const mainNodes = [actionBarNode, contentNode];
+    const mainNodes = [actionBarNode, contentNode, separatorNode];
 
     if (position === "end") {
       mainNodes.reverse();
     }
 
     return <div class={{ [CSS.container]: true }}>{mainNodes}</div>;
   }
+
+  // --------------------------------------------------------------------------
+  //
+  //  private Methods
+  //
+  // --------------------------------------------------------------------------
+
+  setContentWidth(width: number): void {
+    const { contentWidthMax, contentWidthMin, position } = this;
+
+    const rounded = Math.round(position === "end" ? width : width);
+
+    this.contentWidth =
+      typeof contentWidthMax === "number" && typeof contentWidthMin === "number"
+        ? clamp(rounded, contentWidthMin, contentWidthMax)
+        : rounded;
+  }
+
+  updateAriaValues = (): void => {
+    const { contentEl } = this;
+    const computedStyle = contentEl && getComputedStyle(contentEl);
+
+    if (!computedStyle) {
+      return;
+    }
+
+    const max = parseInt(computedStyle.getPropertyValue("max-width"), 10);
+    const min = parseInt(computedStyle.getPropertyValue("min-width"), 10);
+    const valueNow = parseInt(computedStyle.getPropertyValue("width"), 10);
+
+    if (typeof valueNow === "number" && !isNaN(valueNow)) {
+      this.initialContentWidth = valueNow;
+    }
+
+    if (typeof max === "number" && !isNaN(max)) {
+      this.contentWidthMax = max;
+    }
+
+    if (typeof min === "number" && !isNaN(min)) {
+      this.contentWidthMin = min;
+    }
+
+    forceUpdate(this);
+  };
+
+  storeContentEl = (contentEl: HTMLDivElement): void => {
+    this.contentEl = contentEl;
+  };
+
+  getKeyvalue = (event: KeyboardEvent): number => {
+    const key = getKey(event.key);
+    const {
+      step,
+      stepMultiplier,
+      contentWidthMin,
+      contentWidthMax,
+      initialContentWidth,
+      position
+    } = this;
+    const multipliedStep = step * stepMultiplier;
+
+    const MOVEMENT_KEYS = [
+      "ArrowUp",
+      "ArrowDown",
+      "ArrowLeft",
+      "ArrowRight",
+      "Home",
+      "End",
+      "PageUp",
+      "PageDown"
+    ];
+
+    if (MOVEMENT_KEYS.indexOf(key) > -1) {
+      event.preventDefault();
+      event.stopPropagation();
```

Comment:
Do you need to stop propagation of the event? It's not internal and it would break end-user expectations.